### PR TITLE
Fix KubernetesPodOperator xcom push when `get_logs=False`

### LINKED ISSUE #29002: KubernetesPodOperator xcom push failure
### Apache Airflow version

2.5.0

### What happened

Kubernetes pod operator failed to push xcom value.

After upgrading airflow from 2.2.4 to 2.5.0 (and apache-airflow-providers-cncf-kubernetes from 3.0.2 to 5.0.0) pushing of xcom values from kubernetes pod operator stopped working.

### What you think should happen instead

Example of log before the upgrade
```
[2023-01-17 06:57:54,357] {pod_launcher.py:313} INFO - Running command... cat /airflow/xcom/return.json

[2023-01-17 06:57:54,398] {pod_launcher.py:313} INFO - Running command... kill -s SIGINT 1

[2023-01-17 06:57:55,012] {pod_launcher.py:186} INFO - ["No non-accuracy metrics changed more than 10.0% between 2023-01-15 and 2023-01-16\n"]
```
and after the upgrade
```
[2023-01-18T07:12:32.784+0900] {pod_manager.py:368} INFO - Checking if xcom sidecar container is started.
[2023-01-18T07:12:32.804+0900] {pod_manager.py:372} INFO - The xcom sidecar container is started.
[2023-01-18T07:12:32.845+0900] {pod_manager.py:407} INFO - Running command... if [ -s /airflow/xcom/return.json ]; then cat /airflow/xcom/return.json; else echo __airflow_xcom_result_empty__; fi

[2023-01-18T07:12:32.895+0900] {pod_manager.py:407} INFO - Running command... kill -s SIGINT 1

[2023-01-18T07:12:33.405+0900] {kubernetes_pod.py:431} INFO - Result file is empty.
```

Looking at other timestamps in the log file, it appears that the xcom sidecar is run before the pod finishes, instead of waiting until the end.

### How to reproduce

I used `airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator` with `get_logs=False`.

### Operating System

Debian GNU/Linux 11 (bullseye) (based on official airflow image)

### Versions of Apache Airflow Providers

apache-airflow-providers-cncf-kubernetes 3.0.2 and 5.0.0

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Deployed on GKE using the official helm chart

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #29002

---

When  `get_logs=False`, we should wait the `base` container completion before reading the xcom file, to do that we are checking if the container is running, but there is a possibility that the container is in waiting state, and in the current `await_container_completion` method, we consider it as completed.
Instead, I'm checking if the container has the state `terminated` and if not I wait 1s.

### CODE REVIEW

**Comment by potiuk** [comment]
LGTM but I think I need also second opinion (@jedcunningham :) again )? 

I also think having unit tests covering the cases that would have failed previously are needed here. I am always suspicious about changes introducing behavioural change where no tests are touched.

**Review by potiuk — APPROVED** [review]
Approved. @jedcunningham @dstandish ?

**Review by jedcunningham — COMMENTED** [review]
LGTM, but yes we should add test coverage.

File: tests/providers/cncf/kubernetes/utils/test_pod_manager.py [hunk 1]
```diff
@@ -346,6 +351,24 @@ def test_pod_manager_get_client_call_deprecation(self):
...
+                "name": "base",
+                "state.waiting": True if container_state == "waiting" else None,
+                "state.running": True if container_state == "running" else None,
+                "state.terminated": True if container_state == "terminated" else None,
+            }
+        )
+        pod_info = MagicMock()
+        pod_info.status.container_statuses = [container_status]
+        is_terminated = container_is_terminated(pod_info, "base")
+        assert is_terminated == is_terminated
```
Comment (jedcunningham) [inline tests/providers/cncf/kubernetes/utils/test_pod_manager.py]:
```suggestion
        assert container_is_terminated(pod_info, "base") == is_terminated
```

You are overwriting the `is_terminated` you are parameterizing! (I personally like using `expected_` for expectations that are parameterized, but feel free to do it however you'd like)

Comment (hussein-awala) [reply tests/providers/cncf/kubernetes/utils/test_pod_manager.py]: My bad, I just fixed it.

**Review by jedcunningham — APPROVED** [review]

**Comment by hussein-awala** [comment]
@eladkal is it too late to add this fix to cncf.kubernetes release?

**Review by eladkal — APPROVED** [review]

**Comment by eladkal** [comment]
> @eladkal is it too late to add this fix to cncf.kubernetes release?

It will be in next release

**Pull request merged**